pandas/tseries/index.py

```diff
@@ -833,7 +833,7 @@ def astype(self, dtype):
         if dtype == np.object_:
             return self.asobject
         elif dtype == _INT64_DTYPE:
```

Comment:
so this will fail on windows as ``int`` -> ``int32``. BUT we still want the same conversion; so make this

```
elif issubclass(dtype, np.integer):
   ...
```